pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -403,6 +413,173 @@ private SuccessResponse uploadSegment(@Nullable String tableName, TableType tabl
     }
   }
 
+  // Method used to update a list of segments in batch mode with the METADATA upload type.
+  private SuccessResponse uploadSegments(String tableName, TableType tableType,
+      FormDataMultiPart multiParts, boolean enableParallelPushProtection,
+      boolean allowRefresh, HttpHeaders headers, Request request) {
+    String rawTableName = TableNameBuilder.extractRawTableName(tableName);
+    String tableNameWithType = tableType == TableType.OFFLINE
+        ? TableNameBuilder.OFFLINE.tableNameWithType(rawTableName)
+        : TableNameBuilder.REALTIME.tableNameWithType(rawTableName);
+
+    TableConfig tableConfig = _pinotHelixResourceManager.getTableConfig(tableNameWithType);
+    if (tableConfig == null) {
+      throw new ControllerApplicationException(LOGGER, "Failed to fetch table config for table: " + tableNameWithType,
+          Response.Status.BAD_REQUEST);
+    }
+
+    String clientAddress;
+    try {
+      clientAddress = InetAddress.getByName(request.getRemoteAddr()).getHostName();
+    } catch (UnknownHostException ex) {
```

Comment:
(minor) I most of the places we just use `e` to represent exception. Shall we keep the same convention for consistency?